4_basic_assembly/1_signed_operations/answers/2_write_code/0_1.asm

```diff
@@ -12,46 +12,46 @@
 format PE console
 entry start
 
-include 'win32a.inc'
+include 'win32ax.inc'
 
 ; ===============================================
 section '.text' code readable executable
-
 start:
-
-    call    read_hex
-
-    mov     esi, 65536  ; the divisor
-    mov     ecx, eax    ; copy of the user input
-    sub     ebx, ebx    ; temp register to isolate the 8 bit registers in
-    sub     edi, edi    ; answer register
-
-    sub     edx, edx    ; clear mul registers
-    sub     eax, eax    ; clear mul registers
-    mov     al, cl      ; param 1 to eax
-    mov     bl, ch      ; param 2 to ebx
-
-    mul     ebx         ; param 1 * param 2
-
-    mov     edi, eax    ; store the result temp. in edi
-    mov     eax, ecx    ; user input back in eax
-    div     esi
-
-    mov     ecx, eax    ; shifted user input to ebx
-    mov     eax, edi    ; multiplication result back to eax
-    mov     edx, 0      ; clear edx
-    mov     ebx, 0      ; clear ebx
-
-    mov     bl, cl
-    mul     ebx         ; (param 1 * param 2) * param 3
-
-    mov     bl, ch
-    mul     ebx         ; (param 1 * param 2 * param 3) * param 4
-
-    call    print_eax
-
-    ; Exit the process:
-    push    0
-    call    [ExitProcess]
-
+      call read_hex
+      mov edi, eax
+
+      ;right word shift
+      mov esi, 65536
+
+      ;cleared registers for isolating the params
+      mov ebx, 0              
+      mov edx, 0
+      ;result
+      mov ecx, 0
+
+      ;lower word
+      mov bl, al              
+      movzx eax , ah  ;clear eax since it still has part of the input
```

Comment:
@AlexSimpler: Take a look at the diff between this code and the previous one, and you will see that the formatting has changed. The earlier formatting is "in harmony" with the rest of the code in this repository, could you add the relevant spaces to make it easier for the reader?